frontend/pages/SoftwarePage/SoftwareTitleDetailsPage/SoftwareTitleDetailsPage.tsx

```diff
@@ -111,25 +111,34 @@ const SoftwareTitleDetailsPage = ({
         <SoftwareDetailsSummary
           title={softwareTitle.name}
           type={formatSoftwareType(softwareTitle)}
-          versions={softwareTitle.versions.length}
+          versions={softwareTitle.versions?.length ?? 0}
           hosts={softwareTitle.hosts_count}
           queryParams={{ software_title_id: softwareId, team_id: teamIdForApi }}
           name={softwareTitle.name}
           source={softwareTitle.source}
         />
-        <Card
-          borderRadiusSize="large"
-          includeShadow
-          className={`${baseClass}__versions-section`}
-        >
-          <h2>Versions</h2>
-          <SoftwareTitleDetailsTable
-            router={router}
-            data={softwareTitle.versions}
-            isLoading={isSoftwareTitleLoading}
-            teamIdForApi={teamIdForApi}
+        {softwareTitle.hosts_count === 0 ? (
+          <DetailsNoHosts
+            header="Software not detected"
+            details={`No host ${teamIdForApi ? "on this team " : ""}has ${
```

Comment:
@rachaelshaw I know this is as written on Figma, but
"No hosts have [x software] installed."
reads better to me than
"No host has [x software] installed."

Reply:
@RachelElysia that change makes sense to me 👍  (cc @noahtalerman)

Reply:
okay, will update that